digital-workspace/latest/develop/extension-mechanism.md

```diff
@@ -0,0 +1,236 @@
+
+    In the `libs/my-extension` path you will find the following structure:
+
+    * The `src folder` contains all the typescript source code
+    * The `index.ts` file defines all the inclusions of the extension
+    * The `lib/my-extension.module.ts` file defines the module class for the extension
+    * The `README.md` file contains documentation
+    * Other files for testing and configuration
+
+    > **Note:** For more on creating libraries using Nx see [Angular Nx Tutorial - Step 8: Create Libs](https://nx.dev/angular-tutorial/08-create-libs)
+
+3. Once `my-extension` is created, add the configuration to the extension module by editing the `./libs/my-extension/src/lib/my-extension.module.ts` file.
+
+    ```java
+    // Add the following import to the page.
+    import { provideExtensionConfig } from '@alfresco/adf-extensions';
+    // Add providers as described below.
+    NgModule({
+      imports: [CommonModule],
+      providers: [
+        provideExtensionConfig(['my-extension.json'])
+      ]
+    })
+    export class MyExtensionModule {}
+    ```
+
+4. Create a directory called `libs/my-extension/assets`.
+
+    This directory is used to program the extension to add a new item called **hello world** to the **New** button on the landing page of the Digital Workspace.
+
+5. Once created, create a file called `libs/my-extension/assets/my-extension.json` with the following content.
+
+    ```json
+    {
+      "$version": "1.0.0",
+      "$id": "my.extension",
+      "$name": "my adf extension",
+      "$description": "my adf extension",
+      "$license": "Apache-2.0",
+      "actions": [],
+      "features": {
+        "create": [
+          {
+            "id": "my.extension.hello.world",
+            "title": "BYE BYE WORLD! (Logout)",
+            "order": 50,
+            "actions": {
+              "click": "LOGOUT"
+            }
+          }
+        ]
+      },
+      "routes": [],
+      "rules": []
+    }
+    ```
+
+      > **Note:** After the Digital Workspace extension has been created, ensure the `tsconfig.base.json` file includes a link to the `libs/my-extension/src/index.ts` file as part of the paths item. These are default paths that are set during the creation of the extension and it is ufeful to verify the paths in case of any troubleshooting.
+
+6. Edit the `apps/content-ee/src/app/extensions.module.ts` file to add the extension module to the application using the following format.
+
+    ```java
+    // Add the following import to the page.
+    import { MyExtensionModule } from '@alfresco-dbp/my-extension';
+    @NgModule({
+        imports: [
+            ...,
+            MyExtensionModule,
+        ],
+    })
+    export class AppExtensionsModule {}
+    ```
+
+7. Edit the `angular.json` configuration file so the extension is visible from the Digital Workspace app.
+
+8. To the `projects/content-ee/targets/build/options/assets` file add the following.
```

Comment:
This sentence is a continuation from the last one. You need to modify the `angular.json` file adding the following lines to the section `projects/content-ee/targets/build/options/assets` (this is a section inside the angular.json file)

Reply:
Added the two together.